src/GraphQL/Builders/ConnectionBuilder_Typed.cs

```diff
@@ -0,0 +1,291 @@
+using System;
+#if NETSTANDARD2_1
+using System.Diagnostics.CodeAnalysis;
+#endif
+using System.Linq;
+using System.Threading.Tasks;
+using GraphQL.Types;
+using GraphQL.Types.Relay;
+
+#nullable enable
+
+namespace GraphQL.Builders
+{
+    /// <summary>
+    /// Builds a connection field for graphs that have the specified source and return type.
+    /// </summary>
+    public class ConnectionBuilder<TSourceType, TReturnType>
+    {
+        private bool _isBidirectional => FieldType.Arguments.Any(x => x.Name == "before");
+
+        private int? _pageSize
```

Comment:
```suggestion
        private int? PageSize
```

Reply:
Impossible.  Naming conflict.

Reply:
Suggest something else, if you like, and I will implement it.  I don't care.

Reply:
`PageSizeFromMetadata`

Reply:
Ok